### PR TITLE
1806 - ids-toolbar misposition in header overflow menu

### LINKED ISSUE #1806: IDSToolbar: Toolbar in IDSHeader with an IDSAppMenu will misposition the overflow menu
**Describe the bug**
When a page has an app menu, and a toolbar in an IdsHeader, the more actions menu will be mispositioned. Also, a submenu in the more actions menu will not display when hovered over.

**To Reproduce**

1. Unzip [in-header.zip](https://github.com/infor-design/enterprise-wc/files/14089297/in-header.zip) into src/app/components/ids-toolbar/demos in the enterprise-wc-examples/angular-ids-wc. Add the new InHeaderComponent to the two ids-toolbar.module.ts files.
2. Navigate to /ids-toolbar/inheader
3. Click the three line button on the left of the toolbar title. This should open up the application menu
4. Click on the more actions button on the right side of the toolbar.
5. See how the popup menu is moved far to the right and causes a horizontal scrollbar to appear.
6. Hover over the More Options submenu in the more actions menu, and see that no sub menu appears.

**Expected behavior**
The menu is correctly positioned, and doesn't cause a horizontal scrollbar to appear. Submenus should also appear correctly.

**Version**

- ids-enterprise: 17.0.9

**Screenshots**
WCMispositionedOverflow

This is the error in the browser console when the submenu is hovered over:
image

**Platform**
 - Infor Application/Team Name: Landmark
 - Device: (if applicable) All

**Additional context**
Should also probably look at if an action has a very long name. It seems to just expand the popup menu. Should it ellipsis off or wrap the string?

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

Fix so that menus and submenu popup positions are correct when using `IdsToolbar` inside `IdsAppMenu` (and `IdsHeader`)

**Related github/jira issue (required)**:

Closes #1806 

**Steps necessary to review your pull request (required)**:

1. Check out this branch and run: `nvm use && npm install && npm run start`
2. Then go to: http://localhost:4300/ids-toolbar/in-header.html
3. Click the hamburger button on the left of the toolbar title. This should open up the application side-panel menu.
4. Click on the more-actions button (the 3 dots) on the right side of the toolbar.
5. The popup menu should appear properly positioned, and a horizontal scrollbar should not appear.
6. Hover over the "More Options" submenu in the more actions menu, and ensure the submenu appears.

**Included in this Pull Request**:
- [ ] Some documentation for the feature.
- [ ] A test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by tmcconechy — COMMENTED** [review]
@clayinfor the one issue i see is that its loosing the highlight.

Screenshot 2024-05-01 at 10 24 37 AM

The highlight used to stay on the both menu items similar to this:

Screenshot 2024-05-01 at 10 24 21 AM

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Marking as request changes for the above

File: src/components/ids-popup/ids-popup.ts [hunk 1]
```diff
@@ -1379,6 +1379,14 @@ export default class IdsPopup extends Base {
       popupRect = this.onPlace(popupRect);
     }
 
+    if (this.containingElem?.classList?.contains('app-menu-is-open')) {
+      const appMenu = this.containingElem?.querySelector('.app-menu');
+      const appMenuRect = appMenu?.getBoundingClientRect();
+      if (navigator.userAgent.indexOf('Firefox') === -1) {
```
Comment (tmcconechy) [inline src/components/ids-popup/ids-popup.ts:1385]: Does this cause a warning? May also want to sync main here as this PR is now pretty old

**Comment by tmcconechy** [comment]
@clayinfor any updates on this one? Been sitting a long time

**Comment by tmcconechy** [comment]
@clayinfor any updates on this one? Been sitting a long time. Shouldnt need too much to finish it?

**Comment by clayinfor** [comment]
hey @tmcconechy, let's merge this, and create a new task to address the remaining issues.

**Comment by tmcconechy** [comment]
@clayinfor ok great. Can you make the issue as im not sure i know whats left?

**Comment by clayinfor** [comment]
Ok @tmcconechy , I used John's comment to create an issue [here](https://github.com/infor-design/enterprise-wc/issues/2464).

**Pull request merged**